### ISSUE TITLE
Specs broken for Crystal 1.7.0

### ISSUE THREAD

**Issue by manveru** [crystal-ameba/ameba#343]
I'm getting 27 failures:

```
shards build --release --progress --verbose --no-debug
Dependencies are satisfied
Building: ameba
crystal build -o /build/source/bin/ameba src/cli.cr --verbose --release --progress --no-debug
gcc _main.o -o /build/source/bin/ameba  -rdynamic -L/nix/store/gj6rspmx1r7vjdn2amj7vq287bbl982j-boehm-gc-8.2.2/lib -L/nix/store/kxxxzpxww678wwbx78cq5in7gmdnmybq-gmp-with-cxx-6.2.1/lib -L/nix/store/imjp18n6z4r0k6h4w7ckrzkvf3jrhr6k-libevent-2.1.12/lib -L/nix/store/kjg23ai2wb6p3sqpfkdd3xina8ci927z-libxml2-2.9.14/lib -L/nix/store/ml01aw3q1zvs0ncrliv34apl11f0lsx3-libyaml-0.2.5/lib -L/nix/store/nwf2y0nc48ybim56308cr5ccvwkabcqc-openssl-1.1.1q/lib -L/nix/store/l4sxd7mkm9408s8g46alkr1byrw051qq-pcre-8.45/lib -L/nix/store/gidv7w43ilgj22g1mkd3b9if5h4k0xgv-readline-6.3p08/lib -L/nix/store/q7k32ydcqlram7f0l6b1y2c4cs07765y-zlib-1.2.12/lib -L/nix/store/ml01aw3q1zvs0ncrliv34apl11f0lsx3-libyaml-0.2.5/lib -lyaml -lpcre -lm -lgc -lpthread -L/nix/store/imjp18n6z4r0k6h4w7ckrzkvf3jrhr6k-libevent-2.1.12/lib -levent -lrt -lpthread -ldl
buildPhase completed in 2 minutes 4 seconds
running tests
crystal spec
....................................................................................................................................................................................................................................................................*...............................................................F.FF..F.F.FF.F.F..F.F.FF.F..F...F..F........F.F..FF...........................................................................................................................................F.......FF...............................................................................F.......F.......F............................................................................................................................................................................*..........................................................................................................................................................................................................................................................................................................................................................
Pending:
  Ameba::Rule::Style::GuardClause does not report an issue if `else` branch is present but empty
  Ameba::Rule::Lint::LiteralsComparison macro reports in macro scope
Failures:
  1) Ameba::Rule::Performance::SizeAfterFilter reports if there is a select followed by size
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.size
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `select {...}.size`.

       Got:

       [1, 2, 3].select { |e| e > 2 }.size
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `select {...}.size`.
     # spec/ameba/rule/performance/size_after_filter_spec.cr:20
  2) Ameba::Rule::Performance::SizeAfterFilter reports if there is a reject followed by size
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].reject { |e| e < 2 }.size
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `reject {...}.size`.

       Got:

       [1, 2, 3].reject { |e| e < 2 }.size
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `reject {...}.size`.
     # spec/ameba/rule/performance/size_after_filter_spec.cr:33
  3) Ameba::Rule::Performance::SizeAfterFilter reports if a block shorthand used
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].reject(&.empty?).size
               # ^^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `reject {...}.size`.

       Got:

       [1, 2, 3].reject(&.empty?).size
               # ^^^^^^^^^^^^^^^^^^^^^ error: Use `count {...}` instead of `reject {...}.size`.
     # spec/ameba/rule/performance/size_after_filter_spec.cr:40
  4) Ameba::Rule::Performance::SizeAfterFilter reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:2:25"
       Expected: "source.cr:2:25"
            got: "source.cr:2:24"
     # spec/ameba/rule/performance/size_after_filter_spec.cr:74
  5) Ameba::Rule::Performance::MapInsteadOfBlock reports if there is map followed by sum without a block
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       (1..3).map(&.to_u64).sum
            # ^^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`

       Got:

       (1..3).map(&.to_u64).sum
            # ^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`
     # spec/ameba/rule/performance/map_instead_of_block_spec.cr:15
  6) Ameba::Rule::Performance::MapInsteadOfBlock reports if there is map followed by sum without a block (with argument)
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       (1..3).map(&.to_u64).sum(0)
            # ^^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`

       Got:

       (1..3).map(&.to_u64).sum(0)
            # ^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`
     # spec/ameba/rule/performance/map_instead_of_block_spec.cr:28
  7) Ameba::Rule::Performance::MapInsteadOfBlock reports if there is map followed by sum with a block
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       (1..3).map(&.to_u64).sum(&.itself)
            # ^^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`

       Got:

       (1..3).map(&.to_u64).sum(&.itself)
            # ^^^^^^^^^^^^^^^^^ error: Use `sum {...}` instead of `map {...}.sum`
     # spec/ameba/rule/performance/map_instead_of_block_spec.cr:35
  8) Ameba::Rule::Performance::MapInsteadOfBlock reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:1:25"
       Expected: "source.cr:1:25"
            got: "source.cr:1:24"
     # spec/ameba/rule/performance/map_instead_of_block_spec.cr:58
  9) Ameba::Rule::Performance::FlattenAfterMap reports if there is map followed by flatten call
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       %w[Alice Bob].map(&.chars).flatten
                   # ^^^^^^^^^^^^^^^^^^^^^ error: Use `flat_map {...}` instead of `map {...}.flatten`

       Got:

       %w[Alice Bob].map(&.chars).flatten
                   # ^^^^^^^^^^^^^^^^^^^^ error: Use `flat_map {...}` instead of `map {...}.flatten`
     # spec/ameba/rule/performance/flatten_after_map_spec.cr:14
 10) Ameba::Rule::Performance::FlattenAfterMap reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:1:35"
       Expected: "source.cr:1:35"
            got: "source.cr:1:34"
     # spec/ameba/rule/performance/flatten_after_map_spec.cr:43
 11) Ameba::Rule::Performance::FirstLastAfterFilter reports if there is select followed by last
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.last
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `reverse_each.find {...}` instead of `select {...}.last`

       Got:

       [1, 2, 3].select { |e| e > 2 }.last
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `reverse_each.find {...}` instead of `select {...}.last`
     # spec/ameba/rule/performance/first_last_after_filter_spec.cr:18
 12) Ameba::Rule::Performance::FirstLastAfterFilter reports if there is select followed by last?
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.last?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `reverse_each.find {...}` instead of `select {...}.last?`

       Got:

       [1, 2, 3].select { |e| e > 2 }.last?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `reverse_each.find {...}` instead of `select {...}.last?`
     # spec/ameba/rule/performance/first_last_after_filter_spec.cr:31
 13) Ameba::Rule::Performance::FirstLastAfterFilter reports if there is select followed by first
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.first
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `find {...}` instead of `select {...}.first`

       Got:

       [1, 2, 3].select { |e| e > 2 }.first
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `find {...}` instead of `select {...}.first`
     # spec/ameba/rule/performance/first_last_after_filter_spec.cr:38
 14) Ameba::Rule::Performance::FirstLastAfterFilter reports if there is select followed by first?
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.first?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `find {...}` instead of `select {...}.first?`

       Got:

       [1, 2, 3].select { |e| e > 2 }.first?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `find {...}` instead of `select {...}.first?`
     # spec/ameba/rule/performance/first_last_after_filter_spec.cr:51
 15) Ameba::Rule::Performance::FirstLastAfterFilter reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:1:37"
       Expected: "source.cr:1:37"
            got: "source.cr:1:36"
     # spec/ameba/rule/performance/first_last_after_filter_spec.cr:85
 16) Ameba::Rule::Performance::CompactAfterMap reports if there is map followed by compact call
     Failure/Error: expect_issue subject, <<-CRYSTAL
       Expected:

       (1..3).map(&.itself).compact
            # ^^^^^^^^^^^^^^^^^^^^^^ error: Use `compact_map {...}` instead of `map {...}.compact`

       Got:

       (1..3).map(&.itself).compact
            # ^^^^^^^^^^^^^^^^^^^^^ error: Use `compact_map {...}` instead of `map {...}.compact`
     # spec/ameba/rule/performance/compact_after_map_spec.cr:20
 17) Ameba::Rule::Performance::CompactAfterMap reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:1:29"
       Expected: "source.cr:1:29"
            got: "source.cr:1:28"
     # spec/ameba/rule/performance/compact_after_map_spec.cr:49
 18) Ameba::Rule::Performance::AnyAfterFilter reports if there is select followed by any? without a block
     Failure/Error: source = expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].select { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `select {...}.any?`

       Got:

       [1, 2, 3].select { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `select {...}.any?`
     # spec/ameba/rule/performance/any_after_filter_spec.cr:18
 19) Ameba::Rule::Performance::AnyAfterFilter reports if there is reject followed by any? without a block
     Failure/Error: source = expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].reject { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`

       Got:

       [1, 2, 3].reject { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`
     # spec/ameba/rule/performance/any_after_filter_spec.cr:33
 20) Ameba::Rule::Performance::AnyAfterFilter macro reports in macro scope
     Failure/Error: source = expect_issue subject, <<-CRYSTAL
       Expected:

       {{ [1, 2, 3].reject { |e| e > 2  }.any? }}
                  # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`

       Got:

       {{ [1, 2, 3].reject { |e| e > 2  }.any? }}
                  # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`
     # spec/ameba/rule/performance/any_after_filter_spec.cr:61
 21) Ameba::Rule::Performance::AnyAfterFilter reports rule, pos and message
     Failure/Error: source = expect_issue subject, <<-CRYSTAL
       Expected:

       [1, 2, 3].reject { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`

       Got:

       [1, 2, 3].reject { |e| e > 2 }.any?
               # ^^^^^^^^^^^^^^^^^^^^^^^^^ error: Use `any? {...}` instead of `reject {...}.any?`
     # spec/ameba/rule/performance/any_after_filter_spec.cr:71
 22) Ameba::Rule::Lint::EmptyExpression detects empty expression
     Failure/Error: rule.catch(s).should_not be_valid
       Source expected to be invalid, but it is valid.
     # spec/ameba/rule/lint/empty_expression_spec.cr:10
 23) Ameba::Rule::Lint::DuplicatedRequire reports if there are a duplicated requires
     Failure/Error: source = expect_issue subject, <<-CRYSTAL
       Expected:

       require "big"
       require "math"
       require "big"
       # ^{} error: Duplicated require of `big`

       Got:

       require "big"
       require "math"
       require "big"
       # ^^^^^^^^^^^ error: Duplicated require of `big`
     # spec/ameba/rule/lint/duplicated_require_spec.cr:16
 24) Ameba::Rule::Lint::DuplicatedRequire reports rule, pos and message
     Failure/Error: issue.end_location.to_s.should eq ""
       Expected: ""
            got: "source.cr:2:17"
     # spec/ameba/rule/lint/duplicated_require_spec.cr:39
 25) Ameba::Rule::Lint::RedundantWithObject reports rule, location and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:2:30"
       Expected: "source.cr:2:30"
            got: "source.cr:2:29"
     # spec/ameba/rule/lint/redundant_with_object_spec.cr:79
 26) Ameba::Rule::Lint::RedundantWithIndex with_index reports rule, location and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:2:29"
       Expected: "source.cr:2:29"
            got: "source.cr:2:28"
     # spec/ameba/rule/lint/redundant_with_index_spec.cr:80
 27) Ameba::Rule::Lint::RedundantWithIndex each_with_index reports rule, location and message
     Failure/Error: issue.end_location.to_s.should eq "source.cr:2:29"
       Expected: "source.cr:2:29"
            got: "source.cr:2:28"
     # spec/ameba/rule/lint/redundant_with_index_spec.cr:158
Finished in 454.95 milliseconds
1139 examples, 27 failures, 0 errors, 2 pending
```

**Comment by Sija** [comment]
Fixed in #334

**Comment by manveru** [comment]
Thanks :)
